android-core/src/main/kotlin/com.mparticle/audience/BaseAudienceTask.kt

```diff
@@ -0,0 +1,73 @@
+package com.mparticle.audience
+
+import com.mparticle.internal.Logger
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+
+class BaseAudienceTask : AudienceTask<AudienceResponse>() {
+
+    private var isCompleted: Boolean = false
+    private var isSuccessful: Boolean = false
+    private var result: AudienceResponse? = null
+    private var successListeners: MutableSet<AudienceTaskSuccessListener> = java.util.HashSet()
+    private var failureListeners: MutableSet<AudienceTaskFailureListener> = HashSet()
+
+    fun setFailed(errorResponse: AudienceResponse) {
+        isCompleted = true
+        isSuccessful = false
+        try {
+            CoroutineScope(Dispatchers.Main).launch {
+                for (listener in failureListeners) {
+                    listener.onFailure(errorResponse)
```

Comment:
each individual callback invocation should be wrapped in a try/catch so that if one crashes the rest will still be invoked.